avm/src/daemon/daemon.rs

```diff
@@ -133,34 +152,46 @@ async fn control_port(req: Request<Body>) -> Result<Response<Body>, Infallible>
   }
 }
 
-async fn run_agz_b64(agz_b64: &str, priv_key_b64: Option<&str>, cert_b64: Option<&str>) {
-  let bytes = base64::decode(agz_b64).unwrap();
-  let agz = GzDecoder::new(bytes.as_slice());
-  let count = agz.bytes().count();
-  let mut bytecode = vec![0; count / 8];
-  let mut gz = GzDecoder::new(bytes.as_slice());
-  gz.read_i64_into::<LittleEndian>(&mut bytecode).unwrap();
-  if let (Some(priv_key_b64), Some(cert_b64)) = (priv_key_b64, cert_b64) {
-    // Spin up a control port if we can start a secure connection
-    make_server!(
-      HttpType::HTTPS(HttpsConfig {
-        port: 4142, // 4 = A, 1 = L, 2 = N (sideways) => ALAN
-        priv_key_b64: priv_key_b64.to_string(),
-        cert_b64: cert_b64.to_string(),
-      }),
-      control_port
-    );
-    run(
-      bytecode,
-      HttpType::HTTPS(HttpsConfig {
-        port: 443,
-        priv_key_b64: priv_key_b64.to_string(),
-        cert_b64: cert_b64.to_string(),
-      }),
-    )
-    .await;
+async fn run_agz_b64(agz_b64: String, priv_key_b64: Option<String>, cert_b64: Option<String>) {
```

Comment:
can we maybe have this return `Result<String, Box<dyn Error>>` and then panic if there is an error?

Reply:
we could for some cases inside the function return Err but we would not handle the case when `vm::run::run` panics, right?

Reply:
we could check the Error value and panic of specific ones?

Reply:
I think we do want to clean up the AVM to use errors instead of panics at some point, so I think it's better to use `Result`s

Reply:
Ok, as agreed, will update to return `Result` here and later on the `vm::run::run` will be updated to return a `Result` as well.